OpenSim/Simulation/Model/PhysicalFrame.h

```diff
@@ -140,17 +136,16 @@ class OSIMSIMULATION_API PhysicalFrame : public Frame {
         ScaleFactors */
     void scale(const SimTK::Vec3& aScaleFactors);
 
-    /** @name DEPRECATED API 
+    /** Add a Mesh specified by file name to the list of Geometry owned by the ModelComponent.
+    Scale defaults to 1.0 but can be changed on the call line.
+    */
+    void addMeshGeometry(const std::string &aGeometryFileName, const SimTK::Vec3 scale = SimTK::Vec3(1));
```

Comment:
That's a nice service (to use itself as the Frame of the Geometry) that a Frame in general could provide and not only PhysicalFrame. Wouldn't you want that capability for adding any Geometry and not just MeshGeometry? We could provide a virtual `extendAddGeometry()` method that would allow derived Component to do more handling of Geometry that is being added to it. In which case, we should use `Component::addGeometry(const Geometry& geom)` and make a copy especially if we intend to modify it. It would then allow Geometry of one Component (say a Frame) to be  added to another Frame and have its frame connector updated accordingly. It does assume then that the Component owns the Geometry, which I think it does by virtue of `<geometry>` being a list property of the Component.